File: java/test/Ice/facets/DI.java
// **********************************************************************
//
// Copyright (c) 2003-2013 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package test.Ice.facets;

import test.Ice.facets.Test._DDisp;

public final class DI extends _DDisp
{
    public
    DI()
    {
    }

    public String
    callA(Ice.Current current)
    {
        return "A";
    }

    public String
    callB(Ice.Current current)
    {
        return "B";
    }

    public String
    callC(Ice.Current current)
    {
        return "C";
    }

    public String
    callD(Ice.Current current)
    {
        return "D";
    }
}

File: csharp/test/Ice/facets/DI.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.facets;

public sealed class DI : Test.DDisp_
{
    public DI()
    {
    }

    public override string callA(Ice.Current current) => "A";

    public override string callB(Ice.Current current) => "B";

    public override string callC(Ice.Current current) => "C";

    public override string callD(Ice.Current current) => "D";
}
